### PR TITLE
Follow user

### PR DESCRIPTION
**Pull request by vuducdat**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -43,4 +44,16 @@ def authenticated? remember_token
...
+
+  def follow other_user
+    active_relationships.create followed_id: other_user.id
+  end
+
+  def unfollow other_user 
+    active_relationships.find_by(followed_id: other_user.id).destroy
+  end
+
+  def following?other_user
```
Comment (leminhtuan2015) [inline app/models/user.rb]: Space

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -6,6 +6,7 @@
         <li><%= link_to t("title.home2"), root_path %></li>
         <li><%= link_to t("title.help"), help_path %></li>
         <% if logged_in? %>
+          <li><%= link_to "Users", users_path %></li>
```
Comment (leminhtuan2015) [inline app/views/layouts/_header.html.erb]: I18n

File: app/views/shared/_stats.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<% @user ||= current_user %>
+<div class="stats">
+  <a href="<%= following_user_path @user %>">
+    <strong id="following" class="stat">
+      <%= @user.following.count %>
```
Comment (leminhtuan2015) [inline app/views/shared/_stats.html.erb]: It make N+1 query (If u have time u should check it out, if u dont have time i can pass it)

File: app/views/shared/_stats.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    </strong>
+    following
+  </a>
+  <a href="<%= followers_user_path @user %>">
```
Comment (leminhtuan2015) [inline app/views/shared/_stats.html.erb]: link_to

File: app/views/users/_unfollow.html.erb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<%= form_for current_user.active_relationships.find_by followed_id: @user.id,
+  html: { method: :delete }, remote: true) do |f| %>
```
Comment (leminhtuan2015) [inline app/views/users/_unfollow.html.erb]: Convention

File: app/views/shared/_stats.html.erb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<% @user ||= current_user %>
+<div class="stats">
+  <a href="<%= following_user_path @user %>">
+    <strong id="following" class="stat">
+      <%= @user.following.count %>
+    </strong>
+    following
```
Comment (VoTaiTri) [inline app/views/shared/_stats.html.erb]: I18n for text

File: app/views/shared/_stats.html.erb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<% @user ||= current_user %>
+<div class="stats">
+  <a> <%= link_to "", following_user_path(@user) %>
```
Comment (leminhtuan2015) [inline app/views/shared/_stats.html.erb]: Just link_to

File: app/views/shared/_stats.html.erb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    <strong id="following" class="stat">
+      <%= @user.following.count %>
+    </strong>
+    <%= t "title.following" %>
+  </a>
+  <a><%= link_to "", followers_user_path(@user) %>
```
Comment (leminhtuan2015) [inline app/views/shared/_stats.html.erb]: Just link_to

File: app/views/shared/_user_info.html.erb [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<%= link_to gravatar_for(current_user, size: 50), current_user %>
```
Comment (VoTaiTri) [inline app/views/shared/_user_info.html.erb]: store 50 in Setting

File: app/views/shared/_stats.html.erb [hunk 10]
```diff
@@ -0,0 +1,13 @@
+<% @user ||= current_user %>
+<div class="stats">
+  <%= link_to "", following_user_path(@user) %>
+    <strong id="following" class="stat">
```
Comment (leminhtuan2015) [inline app/views/shared/_stats.html.erb]: strong is no child of link_to

```ruby
<strong id="followers" class="stat">
  <%= @user.followers.count %>
</strong>
<%= link_to t("title.follower"), followers_user_path(@user) %>
```

Comment (vuducdat) [inline app/views/shared/_stats.html.erb]: yes bro !

Comment (leminhtuan2015) [inline app/views/shared/_stats.html.erb]: Is strong is child of link_to

File: app/views/shared/_stats.html.erb [hunk 11]
```diff
@@ -0,0 +1,13 @@
+<% @user ||= current_user %>
+<div class="stats">
+  <strong id="followers" class="stat">
+    <%= @user.followers.count %>
+  </strong>
+  <%= link_to t("title.following"), followers_user_path(@user) %>
+  <%= t "title.following" %>
```
Comment (leminhtuan2015) [inline app/views/shared/_stats.html.erb]: Should remove this

**Comment by leminhtuan2015** [comment]
LGTM. well done

File: config/routes.rb [hunk 12]
```diff
@@ -9,9 +9,14 @@
...
   namespace :admin do
     resources :users, only: [:index, :destroy]
   end
+  resources :users do
+    member do
+      get :following, :followers
+    end
+  end
+  resources :relationships, only: [:create, :destroy]
+  resources :users, expect: :destroy
```
Comment (dieunb) [inline config/routes.rb]: `expect`? i think it is `except`

File: app/controllers/users_controller.rb [hunk 13]
```diff
@@ -32,8 +35,32 @@ def update
     end
   end
 
+  def following
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: not follow restfull. You can change last pull request.

File: app/controllers/users_controller.rb [hunk 14]
```diff
@@ -32,8 +35,32 @@ def update
...
+    @user  = User.find params[:id]
+    @users = @user.following.paginate page: params[:page]
+    render "show_follow"
+  end
+
+  def followers
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: not follow restfull. You can change last pull request.

File: config/routes.rb [hunk 15]
```diff
@@ -9,10 +9,15 @@
...
   get "signup"    =>  "users#new"
   
-  resources :users, expect: :destroy
-  
   namespace :admin do
     resources :users, only: [:index, :destroy]
     resources :categories
   end
+  resources :users do
+    member do
```
Comment (dieunb) [inline config/routes.rb]: it is  better using restful

**Pull request merged**